sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosTracerTest.java

```diff
@@ -0,0 +1,375 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+public class CosmosTracerTest extends TestSuiteBase {
+    private static final String ITEM_ID = "tracerDoc";
+    CosmosAsyncClient client;
+    CosmosAsyncDatabase cosmosAsyncDatabase;
+    CosmosAsyncContainer cosmosAsyncContainer;
+
+    @BeforeClass(groups = {"emulator"}, timeOut = SETUP_TIMEOUT)
+    public void beforeClass() {
+        client = new CosmosClientBuilder()
+            .endpoint(TestConfigurations.HOST)
+            .key(TestConfigurations.MASTER_KEY)
+            .directMode(DirectConnectionConfig.getDefaultConfig())
+            .buildAsyncClient();
+        cosmosAsyncDatabase = getSharedCosmosDatabase(client);
+        cosmosAsyncContainer = getSharedMultiPartitionCosmosContainer(client);
+
+    }
+
+    @Test(groups = {"emulator"}, timeOut = TIMEOUT)
+    public void cosmosAsyncClient() {
+        Tracer mockTracer = Mockito.mock(Tracer.class);
+        TracerProvider tracerProvider = Mockito.spy(new TracerProvider(getMockTracer(mockTracer)));
+        ReflectionUtils.setTracerProvider(client, tracerProvider);
+
+        TracerProviderCapture tracerProviderCapture = new TracerProviderCapture();
+        Mockito.doAnswer(tracerProviderCapture).when(tracerProvider).startSpan(Matchers.anyString(),
+            Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+
+        client.createDatabaseIfNotExists(cosmosAsyncDatabase.getId()).block();
+        Context context = tracerProviderCapture.getResult();
+        Mockito.verify(tracerProvider, Mockito.times(1)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+        verifyTracerAttributes(mockTracer, "createDatabaseIfNotExists." + cosmosAsyncDatabase.getId(), context,
+            cosmosAsyncDatabase.getId(), 1);
+
+        client.readAllDatabases(new CosmosQueryRequestOptions()).byPage().single().block();
+        Mockito.verify(tracerProvider, Mockito.times(2)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+        verifyTracerAttributes(mockTracer, "readAllDatabases", context, null, 2);
+
+
+        String query = "select * from c where c.id = '" + cosmosAsyncDatabase.getId() + "'";
+        client.queryDatabases(query, new CosmosQueryRequestOptions()).byPage().single().block();
+        Mockito.verify(tracerProvider, Mockito.times(3)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+        verifyTracerAttributes(mockTracer, "queryDatabases", context, null, 3);
+    }
+
+    @Test(groups = {"emulator"}, timeOut = TIMEOUT)
+    public void cosmosAsyncDatabase() {
+        Tracer mockTracer = Mockito.mock(Tracer.class);
+        TracerProvider tracerProvider = Mockito.spy(new TracerProvider(getMockTracer(mockTracer)));
```

Comment:
Do we need the `spy` if we have a mock of tracer list for a valid tracerProvider object?

nit: Consider adding static import for Mockito methods.

Reply:
we need spy to get the context from startSpan  to verify attributes , and also we are checking on number of timer startSpan is called on provider